src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -219,6 +219,93 @@ MessageHeader header
         }
     }
 
+    [Fact]
+    [OuterLoop]
+    public static void DefaultSettings_Echo_Cookie()
+    {
+        // *** SETUP *** \\
+        var factory = new ChannelFactory<IWcfAspNetCompatibleService>(
+                 new BasicHttpBinding()
+                 {
+                     AllowCookies = true
+                 },
+                 new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+        IWcfAspNetCompatibleService serviceProxy = factory.CreateChannel();
+
+        IHttpCookieContainerManager cookieManager = ((IChannel)serviceProxy).GetProperty<IHttpCookieContainerManager>();
+        Assert.True(cookieManager != null);
+        Assert.True(cookieManager.CookieContainer != null);
+        string cookieName = "cookieName";
+        string cookieValue = "cookieValue";
+        string cookieSentOut = $"{cookieName}={cookieValue}";
+        cookieManager.CookieContainer.Add(new Uri(Endpoints.HttpBaseAddress_Basic), new System.Net.Cookie(cookieName, cookieValue));
+
+        try
+        {
+            // *** EXECUTE *** \\
+            string cookieSentBack = serviceProxy.EchoCookie();
+
+            // *** VALIDATE *** \\
+            Assert.True(cookieSentOut == cookieSentBack,
```

Comment:
Is System.Net.Cookie comparable directly? If so, consider using that instead of using a string compare

---
In reply to: [55325316](https://github.com/dotnet/wcf/pull/883#discussion_r55325316) [](ancestors = 55325316)